scripts/package_sign_tacos.sh

```diff
@@ -0,0 +1,139 @@
+#
+# Parameters passed to script:
+#   1) Build number (required)
+#   2) Jenkins Workspace (required)
+#   3) Branch name (optional)
+#
+# Requirements prior to running script:
+#   - Digicert Software Trust Manager is installed and configured.
+#   - The command "smctl healthcheck" runs successfully, reporting:
+#     - A valid signer certificate is available.
+#     - The user has the privilege to sign.
+#
+
+# Verify that build number was passed in
+if [ -z $1 ]; then
+   echo Missing build number parameter
+   exit 1
+fi
+
+BUILD=$1
+
+# Verify that Jenkins workspace directorywas passed in
+if [ -z $2 ]; then
+   echo Missing workspace directory parameter
+   exit 1
+fi
+
+WKSP=$JENKINS_HOME/$2
+echo Using Jenkins workspace: $WKSP
+
+# Check if branch name was passed
+if [ $3 ]; then
+   BRANCH="-b $3"
+fi
+echo DEBUG BRANCH=$BRANCH
+
+# Check prerequisites
+if [ -z $(which jarsigner) ]; then
+   echo jarsigner not found
+   exit 1
+fi
+
+# Set directory variables
+BASEDIR=$HOME/tableau
+TACODIR=$BASEDIR/tacos
+LOGDIR=$BASEDIR/logs
+SCRIPTDIR=$BASEDIR/scripts
+GITHUBDIR=$BASEDIR/github
+ACTIANDIR=$GITHUBDIR/actian_tableau_connector
+JDBCDIR=$ACTIANDIR/actian_jdbc
+ODBCDIR=$ACTIANDIR/actian_odbc
+TABSDKDIR=$GITHUBDIR/connector-plugin-sdk
+PKGRDIR=$GITHUBDIR/connector-plugin-sdk/connector-packager
+
+# Ensure subdirectories exist
+if [ ! -d "$TACODIR" ]; then
+  mkdir -p $TACODIR/unsigned
+fi
+if [ ! -d "$LOGDIR" ]; then
+  mkdir $LOGDIR
+fi
+
+# Set repo variables
+TABSDKREPO=https://github.com/tableau/connector-plugin-sdk.git
+ACTIANREPO=https://github.com/ActianCorp/actian_tableau_connector.git
+
+TS=$(date +"%Y%b%d-%H%M%S")
+
+# Log parameters passed to script
+echo $TS  $BUILD  $WKSP >> $LOGDIR/builds.log
+
+cd $BASEDIR
+
+# Create Python vitual environment
+python -m venv .venv
+. ./.venv/bin/activate
+
+# Clone repositories#
+rm -rf $GITHUBDIR/*
+git clone $BRANCH $ACTIANREPO $ACTIANDIR
+git clone $TABSDKREPO $TABSDKDIR
+
+# Get version from manifest.xml
+VERJDBC=$(grep plugin-version $JDBCDIR/manifest.xml  | cut -d '=' -f4 | cut -d "'" -f2)
+VERODBC=$(grep plugin-version $ODBCDIR/manifest.xml  | cut -d '=' -f4 | cut -d "'" -f2)
+
+# Set taco file names
+JDBC_TACO=actian_jdbc-v$VERJDBC.taco
+ODBC_TACO=actian_odbc-v$VERODBC.taco
+JDBC_TACO_FINAL=$JDBC_TACO.build_$BUILD
+ODBC_TACO_FINAL=$ODBC_TACO.build_$BUILD
+
+# Package tacos
+cd $TABSDKDIR/connector-packager
+python setup.py install
+python -m connector_packager.package $JDBCDIR  --dest $TACODIR -l $LOGDIR
+mv $LOGDIR/packaging_logs.txt $LOGDIR/packaging_logs_jdbc_build_$BUILD.txt
+python -m connector_packager.package $ODBCDIR  --dest $TACODIR -l $LOGDIR
+mv $LOGDIR/packaging_logs.txt $LOGDIR/packaging_logs_odbc_build_$BUILD.txt
+
+cd $BASEDIR
+
+# Set environment variables needed for accessing certificate
+export SM_HOST=https://clientauth.one.digicert.com
+export SM_CLIENT_CERT_FILE=/usr/local/digicert/certs/Certificate_pkcs12.p12
```

Comment:
also move higher, could be optional with default if omitted.

Reply:
Moved variable initialization to top of script (after comments).